lib/widget/roles_selection.dart

```diff
@@ -0,0 +1,211 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_redux/flutter_redux.dart';
+import 'package:heist/app_localizations.dart';
+import 'package:heist/colors.dart';
+import 'package:heist/middleware/room_middleware.dart';
+import 'package:heist/role.dart';
+import 'package:heist/selectors/selectors.dart';
+import 'package:heist/state.dart';
+import 'package:heist/widget/common.dart';
+import 'package:redux/redux.dart';
+
+class RolesSelection extends StatefulWidget {
+  final Store<GameModel> _store;
+
+  RolesSelection(this._store);
+
+  @override
+  State<StatefulWidget> createState() => RolesSelectionState(_store);
+}
+
+class RolesSelectionState extends State<RolesSelection> {
+  final Store<GameModel> _store;
+
+  RolesSelectionState(this._store);
+
+  @override
+  Widget build(BuildContext context) =>
+      new StoreConnector<GameModel, RolesSelectionModel>(
+          converter: (store) => new RolesSelectionModel._(
+              store.state.room.numPlayers,
+              amOwner(store.state),
+              store.state.room.roles),
+          distinct: true,
+          builder: (context, viewModel) {
+            return _choosingRoles(
+                context, _store, viewModel.numPlayers, viewModel.amOwner);
+          });
+
+  Widget _choosingRoles(final BuildContext context,
+      final Store<GameModel> store, final int numPlayers, final bool amOwner) {
+    final Set<Role> defaultRoles = Roles.numPlayersToRolesMap[numPlayers];
+    final int numFriendlyRoles =
+        _getNumRolesInTeam(Roles.getRoleIds(defaultRoles), Team.FRIENDLY);
+    final int numScaryRoles =
+        _getNumRolesInTeam(Roles.getRoleIds(defaultRoles), Team.SCARY);
+
+    final Set<String> currentRoleIds = store.state.room.roles;
```

Comment:
Can you pass these through from the `viewModel`?

Reply:
Yep, they're in the viewModel already